R/rvar-slice.R

```diff
@@ -125,7 +125,20 @@
   ))
 
   if (drop) {
-    drop_(x)
+    .dim <- dim(x)
+    if (length(.dim) > 1) {
+      # with exactly 1 dimension left we don't want to drop anything
+      # (otherwise names get lost), so only do this with > 1 dimension
+      keep_dim <- .dim != 1
+      .dimnames <- dimnames(x)
+      dim(x) <- .dim[keep_dim]
+      # for comparison / testing, ensure if no dimnames have names that we
+      # actually have those names be NULL (rather than just empty strings)
+      new_dimnames <- .dimnames[keep_dim]
+      if (all(names(new_dimnames) == "")) names(new_dimnames) <- NULL
+      dimnames(x) <- new_dimnames
+    }
```

Comment:
> There was a separate drop_() helper within rvar-slice tools. I first tried to merge both functions but it wasn't clean due to rvar objects not being of array class and having some edge cases. Then I realized drop_() was a small function that was getting called only once, so I got rid of it to prevent confusion (assuming it is not used in any downstream packages as rvar already has a slicer method with drop option).

That's fine with me. I had thought I would need that function in more than one spot in rvar internals and then I didn't. Stylistically I would probably change this:

```r
    x
  } else {
    x
  }
```

to this:

```r
  }

  x
```

Reply:
Updated it, thanks! Let me know if there is any other change you'd like.

Reply:
Thanks!